### DISCUSSION TITLE
Cloudflare Pages Cron and Queues

### DISCUSSION THREAD

**Discussion by bringking** [remix-run/remix#8138] [Q&A]
To take full advantage of Cloudflare, workers can export other handlers besides fetch for things like consuming a Queue message, or getting triggered by a Cron. The docs show something like this - 

```
export default {
  async fetch(request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
  },
  async queue(batch: MessageBatch<any>, env: Env): Promise<void> {
  },
  async scheduled(event, env, ctx) {
  },
}
```
With Remix/Cloudflare, it makes sense to let the Remix build handle `fetch`, but how can we also specify handlers for the other entrypoints? I'm specifically using the Remix Cloudflare Pages runtime

**Comment by tavoyne** [comment]
Did you find a workaround, @bringking?

I'm also super interested in being able to schedule cron jobs in a Remix + CF Pages app.

**Comment by hi-ogawa** [comment]
I think what's your asking is a limitation of Cloudflare Pages in general.
Essentially Cloudflare pages uses Cloudflare workers for their "Function", but I think they don't expose all the feature from workers to pages. https://developers.cloudflare.com/pages/platform/functions/api-reference/ Also from their point of view, it doesn't necessary make sense to say "handle `queue` in this function" since their function works based on http request route.

If you use Cloudflare Workers directly, then I believe it's just a matter adding an extra code for `queue, scheduled` handler etc... to existing template around here:

https://github.com/remix-run/remix/blob/1a8a5216106bd8c3073cc3e5e5399a32c981db74/templates/cloudflare-workers/server.ts#L14-L15

While looking around their doc, I found this "Advanced mode" https://developers.cloudflare.com/pages/platform/functions/advanced-mode/ which seems to give you a full control over worker script. This might allow setting up `queue` handler etc... but I haven't tried.
(Note that Remix's current template use `functions/[[path]].js` to handle all requests as "function" which is not "advanced mode")

Either way, I think this kind of question might be better asked on cloudflare/wrangler side since people here might not be familiar with this subtle nuance.

---

EDIT: Extra note. It seems sveltekit uses "advanced mode" (`_worker.js` instead of `functions/[[path]].js`) for their cloudflare adapter, but the worker code is generated by adapter anyways, so I don't think users have control over it either https://github.com/sveltejs/kit/blob/3d98c37c09aa53d1bf8e5dcf12aef62a43afcae8/packages/adapter-cloudflare/index.js#L41

**Comment by candidia** [comment]
I'm also curious about this because the vercel remix integration allows making any route a cron if I remember right.

**Comment by hyusetiawan** [comment]
im curious to know how to accomplish this but it seems there is no way to do this currently?
